CONTRIBUTING.md

```diff
@@ -1,12 +1,78 @@
+  
+**Before writing code**
+
+  This can save you a lot of time. We've seen PRs where customers solve an issue in a way which either wouldn't fit into upgrade-assistant because of how it's designed or would change upgrade-assistant in a way which is not something we'd like to do. To avoid these situations, we encourage customers to discuss the preferred design with the team first. To do so, file a new `design proposal` issue, link to the issue you'd like to address and provide detailed information about how you'd like to solve a specific problem. We triage issues periodically and it will not take long for a team member to engage with you on that proposal.
+  When you get an agreement from our team members that the design proposal you have is solid, then go ahead and prepare the PR.
+  To file a design proposal, look for the relevant issue in the `New issue` page or simply click [this link](https://github.com/dotnet/upgrade-assistant/issues/new?assignees=&labels=design-proposal&template=10_design_proposal.md):
+  ![image](https://user-images.githubusercontent.com/34246760/107969904-41b9ae80-6f65-11eb-8b84-d15e7d94753b.png)
+  
+**Before submitting the pull request**
+
+Before submitting a pull request, make sure that it checks the following requirements:
+
+- You find an existing issue with the "help-wanted" label or discuss with the team to agree on adding a new issue with that label
+- You post a high-level description of how it will be implemented, and receive a positive acknowledgement from the team before getting too committed to the approach or investing too much effort implementing it
+- You add test coverage following existing patterns within the codebase
+- Your code matches the existing syntax conventions within the codebase
+- Your PR is small, focused, and avoids making unrelated changes
+  
+If your pull request contains any of the below, it's less likely to be merged:
+
+- Changes that break backward compatibility
+- Changes that are only wanted by one person/company. Changes need to benefit a large enough portion of upgrade-assistant users.
+- Changes that add entirely new feature areas without prior agreement
+- Changes that are mostly about refactoring existing code or code style
+- Very large PRs that would take hours to review (remember, we're trying to help lots of people at once). For larger work areas, please discuss with us to find ways of breaking it down into smaller, incremental pieces that can go into separate PRs.
+
+**During pull request review**
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
+
+### Resources to help you get started
```

Comment:
sounds very similar to the ** Finding an issue to work on ** section. do we need both? or did I miss the intent?

Reply:
The intent is to give a TLDR list of links. I think it will be useful to have this short list.